### DISCUSSION TITLE
Github like `diff` extension to compare two versions

### DISCUSSION THREAD

**Discussion by piszczu4** [ueberdosis/tiptap#4800] [General Questions]
Is there any official/community extension which enable the comparison of two tiptap docs? I mean either by comparing the rendered document and wrapping removed/added nodes in green/red outlines, or by comparing two `html`s (or `json`) outputs, sth similar to what github do? 

Github's editor has the `edit` history and highlight changes as below where either text is highlighted or there is a left green border.
![image](https://github.com/ueberdosis/tiptap/assets/88668060/c1bebe9e-8104-4a6c-b31f-1c0a07678a6f)

Any suggestions how to obtain simialr effect?

**EDIT**
[here](https://tiptap.dev/blog/release-notes/tiptap-collaboration-history)  @philipisik wrote

![image](https://github.com/ueberdosis/tiptap/assets/88668060/186e4925-0194-44d2-930f-1951e997866e)

Does anyone know what is the status?

**Comment by philipisik** [comment]
Indeed, the diff view is on our Q1 roadmap and we're working on a POC. Things escalated a bit because a diff view in combination with version history was more complex to solve than we thought 😅

**Reply by piszczu4** [reply]
@philipisik thank you for the update, hope to see this feature, good luck!

**Reply by spaceemotion** [reply]
Are there any updates on this? If not, it seems I have to build my own as https://github.com/ProseMirror/prosemirror-changeset seems to have too many quirks..

**Reply by nperez0111** [reply]
Yea I am still working on it. I have basically had to rewrite that library to get it to work properly.

**Reply by piszczu4** [reply]
> Yea I am still working on it. I have basically had to rewrite that library to get it to work properly.

And how is it going? Are we closer to get this extension?

**Reply by nperez0111** [reply]
Extension is done. We are working some of our existing customers to make sure it is ready for prime time. If you want to be one of them please reach out to humans@tiptap.dev with the same email address as your tiptap account
